urld0f681c558_farom57_AstroidDriver_tgz-pJ8-laazotea_indi_server_examples_MinimumINDIServerJ8/src/laazotea/indi/server/examples/INDIBasicServer.java

```diff
@@ -0,0 +1,526 @@
+    System.err.println("  -help                Shows this help.");
+    System.err.println("  -p=port              Port to which the Server will listen.");
+    System.err.println("  -add=jarFile         Loads all INDIDrivers in the jarFile.");
+    System.err.println("  -addn=driverPath     Loads the native driver described by driverPath.");
+    System.err.println("  -connect=host[:port] Loads the drivers in a remote INDI server.\n");
+  }
+
+  /**
+   * Prints some help about the possible commands of the program to the error
+   * stream.
+   */
+  private static void printCommandHelp() {
+    System.err.println("\nThe following commands can be used:");
+    System.err.println("  help                    Shows this help.");
+    System.err.println("  list                    Lists all loaded drivers.");
+    System.err.println("  add jarFile             Loads all INDIDrivers in the jarFile.");
+    System.err.println("  remove jarFile          Removes all INDIDrivers in the jarFile");
+    System.err.println("  reload jarFile          Reloads all INDIDrivers in the jarFile");
+    System.err.println("  r                       Reloads all INDIDrivers in jarFiles (useful for testing)");
+    System.err.println("  addn driverPath         Loads the native driver described by driverPath");
+    System.err.println("  removen driverPath      Removes the native driver described by driverPath");
+    System.err.println("  reloadn driverPath      Reloads the native driver described by driverPath");
+    System.err.println("  connect host[:port]     Loads the drivers in a remote INDI server.");
+    System.err.println("  disconnect host[:port]  Removes the drivers in a remote INDI server.");
+    System.err.println("  stop                    Stops the Server and breaks all Client connections.");
+    System.err.println("  start                   Starts the Server (if it was previously not).\n");
+  }
+
+  /**
+   * Splits an argument in two. Separator char is =
+   *
+   * @param arg The argument to split.
+   * @return An array of the two components of the parameter.
+   */
+  private static String[] splitArgument(String arg) {
+    int pos = arg.indexOf("=");
+
+    if (pos != -1) {
+      return new String[]{arg.substring(0, pos), arg.substring(pos + 1)};
+    } else {
+      return new String[]{arg, ""};
+    }
+  }
+
+  /**
+   * Parses an input command. Possible commands are
+   * <code>help</code>,
+   * <code>list</code>,
+   * <code>add jarFile</code>,
+   * <code>remove jarFile</code>,
+   * <code>reload jarFile</code>,
+   * <code>r</code>,
+   * <code>addn driverPath</code>,
+   * <code>removen driverPath</code>,
+   * <code>reloadn driverPath</code>,
+   * <code>connect host[:port]</code>,
+   * <code>disconnect host[:port]</code>,
+   * <code>stop</code> and
+   * <code>start</code>.
+   *
+   * @param line The line to be parsed.
+   */
+  private static void parseInputLine(String line) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 28. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 368-398 into a separate method.